src/main/java/gtPlusPlus/core/item/base/BaseItemComponent.java

```diff
@@ -359,100 +366,550 @@ else if (this.componentMaterial.getRGBA()[3] == 3) {
 						((Map<Integer, Short[]>) extraData).put(aSlot++, new Short[] {  0, 255,   0});
 						((Map<Integer, Short[]>) extraData).put(aSlot++, new Short[] {  0, 255,   0});
 						((Map<Integer, Short[]>) extraData).put(aSlot++, new Short[] {  0, 255,   0});
+						((Map<Integer, Short[]>) extraData).put(aSlot++, new Short[] {  0, 255,   0});
```

Comment:
Do you have to repeat these lines over and over? How about some good old for loops?